src/main/httpCommandServer.ts

```diff
@@ -41,18 +52,35 @@ export default class HttpCommandServer {
     console.log('CLI server is now ready.');
   }
 
-  shutdown() {
-    this.server.close();
-  }
-
   protected handleRequest(request: http.IncomingMessage, response: http.ServerResponse) {
-    if (!this.basicAuth(request.headers.authorization ?? '')) {
-      response.writeHead(401, { 'Content-Type': 'text/plain' });
-    } else {
-      response.writeHead(200, { 'Content-Type': 'text/plain' });
-      response.write('Nothing to see here yet.');
+    try {
+      if (!this.basicAuth(request.headers.authorization ?? '')) {
+        response.writeHead(401, { 'Content-Type': 'text/plain' });
+
+        return;
+      }
+      const method = request.method;
+      const url = new URL(request.url as string, `http://${ request.headers.host }`);
+      const path = url.pathname;
+      const pathParts = path.split('/');
+      const commandName = pathParts[0] || pathParts[1];
+      // TODO: Further processing of path parts, query parameters, and -d payload to be done later.
```

Comment:
"-d playload" is presumably "request body".

Reply:
Thank you. Did I mention somewhere that I won't be surprised if at some point it might be better to replace this homegrown code with Express, or a similar module? This comment makes me think that day is closer than we think.